src/client/ClientDataResolver.js

```diff
@@ -317,6 +317,66 @@ class ClientDataResolver {
     }
     return null;
   }
+
+  /**
+   * Can be a Hex Literal, Hex String, Number, RGB Array, or one of the following
+   * ```
+   * [
+   *   'DEFAULT',
+   *   'AQUA',
+   *   'GREEN',
+   *   'BLUE',
+   *   'PURPLE',
+   *   'GOLD',
+   *   'ORANGE',
+   *   'RED',
+   *   'GREY',
+   *   'DARKER_GREY',
+   *   'NAVY',
+   *   'DARK_AQUA',
+   *   'DARK_GREEN',
+   *   'DARK_BLUE',
+   *   'DARK_PURPLE',
+   *   'DARK_GOLD',
+   *   'DARK_ORANGE',
+   *   'DARK_RED',
+   *   'DARK_GREY',
+   *   'LIGHT_GREY',
+   *   'DARK_NAVY',
+   * ]
+   * ```
+   * @typedef {String|number|Array} ColorResolvable
+   */
+
+  /**
+   * @param {ColorResolvable} color Color to resolve
+   * @returns {number} A color
+   */
+  static resolveColor(color) {
+    if (typeof color === 'string') {
+      if (color in Constants.Colors) {
+        color = Constants.Colors[color];
+      } else {
+        color = parseInt(color.replace('#', ''), 16);
+      }
```

Comment:
357-361 could just be simplified to `color = Constants.Colors[color] || parseInt(color.replace('#', ''), 16);`